zipline/gens/brokers/rh/robinhood.py

```diff
@@ -0,0 +1,1147 @@
+"""Robinhood.py: a collection of utilities for working with Robinhood's Private API """
+
+#Standard libraries
+import logging
+import warnings
+
+from enum import Enum
+
+#External dependencies
+from six.moves.urllib.parse import unquote
+from six.moves.urllib.request import getproxies
+from six.moves import input
+
+import getpass
+import requests
+import six
+
+#Application-specific imports
+from zipline.gens.brokers.rh import rh_exceptions as RH_exception
+
+
+class Bounds(Enum):
+    """Enum for bounds in `historicals` endpoint """
+
+    REGULAR = 'regular'
+    EXTENDED = 'extended'
+
+
+class Transaction(Enum):
+    """Enum for buy/sell orders """
+
+    BUY = 'buy'
+    SELL = 'sell'
+
+
+class Robinhood:
+    """Wrapper class for fetching/parsing Robinhood endpoints """
+
+    endpoints = {
+        "login": "https://api.robinhood.com/api-token-auth/",
+        "logout": "https://api.robinhood.com/api-token-logout/",
+        "investment_profile": "https://api.robinhood.com/user/investment_profile/",
+        "accounts": "https://api.robinhood.com/accounts/",
+        "ach_iav_auth": "https://api.robinhood.com/ach/iav/auth/",
+        "ach_relationships": "https://api.robinhood.com/ach/relationships/",
+        "ach_transfers": "https://api.robinhood.com/ach/transfers/",
+        "applications": "https://api.robinhood.com/applications/",
+        "dividends": "https://api.robinhood.com/dividends/",
+        "edocuments": "https://api.robinhood.com/documents/",
+        "instruments": "https://api.robinhood.com/instruments/",
+        "margin_upgrades": "https://api.robinhood.com/margin/upgrades/",
+        "markets": "https://api.robinhood.com/markets/",
+        "notifications": "https://api.robinhood.com/notifications/",
+        "orders": "https://api.robinhood.com/orders/",
+        "password_reset": "https://api.robinhood.com/password_reset/request/",
+        "portfolios": "https://api.robinhood.com/portfolios/",
+        "positions": "https://api.robinhood.com/positions/",
+        "quotes": "https://api.robinhood.com/quotes/",
+        "historicals": "https://api.robinhood.com/quotes/historicals/",
+        "document_requests": "https://api.robinhood.com/upload/document_requests/",
+        "user": "https://api.robinhood.com/user/",
+        "watchlists": "https://api.robinhood.com/watchlists/",
+        "news": "https://api.robinhood.com/midlands/news/",
+        "fundamentals": "https://api.robinhood.com/fundamentals/",
+    }
+
+    session = None
+    username = None
+    password = None
+    headers = None
+    auth_token = None
+
+    logger = logging.getLogger('Robinhood')
```

Comment:
Could you please move this out to the top level of the module? Additionally, please use logbook to instantiate - that's the logger used throughout zipline code.